contracts/extensions/votingReputation/VotingReputationStaking.sol

```diff
@@ -0,0 +1,240 @@
+    Motion storage motion = motions[_motionId];
+
+    uint256 totalSideStake = motion.stakes[_vote] + motion.pastVoterComp[_vote];
+    if (totalSideStake == 0) { return (0, 0); }
+
+    uint256 stakeFraction = wdiv(stakes[_motionId][_staker][_vote], totalSideStake);
+
+    uint256 realStake = wmul(stakeFraction, motion.stakes[_vote]);
+
+    uint256 stakerReward;
+    uint256 repPenalty;
+
+    // Went to a vote, use vote to determine reward or penalty
+    if ((motion.votes[NAY] + motion.votes[YAY]) > 0) {
+
+      uint256 loserStake;
+      uint256 winnerStake;
+      if (motion.votes[YAY] > motion.votes[NAY]){
+        loserStake = motion.stakes[NAY];
+        winnerStake = motion.stakes[YAY];
+      } else {
+        loserStake = motion.stakes[YAY];
+        winnerStake = motion.stakes[NAY];
+      }
+
+      loserStake -= motion.paidVoterComp;
+      uint256 totalVotes = motion.votes[NAY] + motion.votes[YAY];
+      uint256 winFraction = wdiv(motion.votes[_vote], totalVotes);
+      uint256 winShare = wmul(winFraction, 2 * WAD); // On a scale of 0-2 WAD
+
+      if (winShare > WAD || (winShare == WAD && _vote == NAY)) {
+        // 50% gets 0% of loser's stake, 100% gets 100% of loser's stake, linear in between
+        stakerReward = wmul(stakeFraction, (winnerStake + wmul(loserStake, winShare - WAD)));
+      } else {
+        stakerReward = wmul(stakeFraction, wmul(loserStake, winShare));
+        repPenalty = realStake - stakerReward;
+      }
+
+    // Determine rewards based on stakes alone
+    } else {
+      assert(motion.paidVoterComp == 0);
+      uint256 requiredStake = getRequiredStake(_motionId);
+
+      // Your side fully staked, receive 10% (proportional) of loser's stake
+      if (
+        motion.stakes[_vote] == requiredStake &&
+        motion.stakes[flip(_vote)] < requiredStake
+      ) {
+
+        uint256 loserStake = motion.stakes[flip(_vote)];
+        uint256 totalPenalty = wmul(loserStake, WAD / 10);
+        stakerReward = wmul(stakeFraction, (requiredStake + totalPenalty));
+
+      // Opponent's side fully staked, pay 10% penalty
+      } else if (
+        motion.stakes[_vote] < requiredStake &&
+        motion.stakes[flip(_vote)] == requiredStake
+      ) {
+
+        uint256 loserStake = motion.stakes[_vote];
+        uint256 totalPenalty = wmul(loserStake, WAD / 10);
+        stakerReward = wmul(stakeFraction, loserStake - totalPenalty);
+        repPenalty = realStake - stakerReward;
+
+      // Neither side fully staked (or no votes were revealed), no reward or penalty
+      } else {
+
+        stakerReward = realStake;
+
+      }
+    }
+
+    return (stakerReward, repPenalty);
+  }
+
+  // Internal
+
+  function lockExpenditure(uint256 _motionId) internal {
+    Motion storage motion = motions[_motionId];
+    bytes memory action = getExpenditureAction(motion.action);
+    uint256 expenditureId = getExpenditureId(action);
```

Comment:
I did initially, but I thought it would be a suboptimal implementation, since some of that data is already stored on the motion struct (i.e. the `domainId`). I figured it would be easier to look up the `expenditureId` the few times it's needed, and so wouldn't be worth storing.